azure-client-runtime/src/main/java/com/microsoft/azure/LocationPollStrategy.java

```diff
@@ -64,17 +69,41 @@ public boolean isDone() {
      * Try to create a new LocationOperationPollStrategy object that will poll the provided location
      * URL. If the provided HttpResponse doesn't have a Location header or the header is empty,
      * then null will be returned.
-     * @param fullyQualifiedMethodName The fully qualified name of the method that initiated the
-     *                                 long running operation.
+     * @param originalHttpRequest The original HTTP request.
      * @param httpResponse The HTTP response that the required header values for this pollStrategy
      *                     will be read from.
      * @param delayInMilliseconds The delay (in milliseconds) that the resulting pollStrategy will
      *                            use when polling.
      */
-    static PollStrategy tryToCreate(RestProxy restProxy, String fullyQualifiedMethodName, HttpResponse httpResponse, long delayInMilliseconds) {
+    static PollStrategy tryToCreate(RestProxy restProxy, HttpRequest originalHttpRequest, HttpResponse httpResponse, long delayInMilliseconds) {
         final String locationUrl = httpResponse.headerValue(HEADER_NAME);
-        return locationUrl != null && !locationUrl.isEmpty()
-                ? new LocationPollStrategy(restProxy, fullyQualifiedMethodName, locationUrl, delayInMilliseconds)
-                : null;
+
+        String pollUrl = null;
+        if (locationUrl != null && !locationUrl.isEmpty()) {
+            if (locationUrl.startsWith("/")) {
```

Comment:
can consider using https://stackoverflow.com/questions/8273019/how-to-merge-two-uris-in-java#8273086 to combine a base URL with a relative URL

Reply:
Done.